ipie/utils/from_trexio.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 The ipie Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors: Anthony Scemama <scemama@irsamc.ups-lse.fr>
+#
+
+"""Generate AFQMC data from a TREXIO file."""
+try:
+  import trexio
+except:
+  print("TREXIO is not installed. Try pip install trexio.")
+  raise
+
+import time
+from dataclasses import dataclass
+from typing import Tuple, Union
+
+import h5py
+import numpy
+import scipy.linalg
+from pyscf import lib, scf
+
+from ipie.estimators.generic import core_contribution_cholesky
+from ipie.legacy.estimators.generic import local_energy_generic_cholesky
+from ipie.legacy.estimators.greens_function import gab
+from ipie.utils.io import write_hamiltonian, write_wavefunction
+from ipie.utils.misc import dotdict
+
+
+def gen_ipie_from_trexio(
+    trexio_filename: str,
+    verbose: bool = True
+) -> None:
+
+    trexio_file = trexio.File(trexio_filename, 'r', trexio.TREXIO_AUTO)
+
+    mo_num   = trexio.read_mo_num(trexio_file)
+    if verbose:
+      print(f"mo_num = {mo_num} # MOs")
+
+    chol_num = trexio.read_mo_2e_int_eri_cholesky_num(trexio_file)
+    if verbose:
+      print(f"chol_num = {chol_num} # Cholesky vectors")
+
+    nup = trexio.read_electron_up_num(trexio_file)
+    ndn = trexio.read_electron_dn_num(trexio_file)
+    if verbose:
+      print(f"{nup} up- and {ndn} down-spin electrons")
+
+    hcore = trexio.read_mo_1e_int_core_hamiltonian(trexio_file)
+    if verbose:
+      print(f"Read core Hamiltonian")
+
+    ndet = trexio.read_determinant_num(trexio_file)
+    if verbose:
+      print(f"{ndet} determinants")
+
+    ci_coeffs = trexio.read_determinant_coefficient(trexio_file, 0, ndet)[0]
```

Comment:
Does trexio use ababab ordering or aaabbb ordering for determinants? If it's ababab we might need to have to change the phase similar to what is done for dice.

Reply:
aaabbb.
we checked with Joohno, and using a CIPSI wave function coming from Quantum Package gave the correct energy for the trial wave function, and a very good estimate of the FCI energy for the AF-QMC result :-)